packages/react-composites/src/composites/CallComposite/adapter/AzureCommunicationCallAdapter.ts

```diff
@@ -879,23 +907,52 @@ export const useAzureCommunicationCallAdapter = (
   return adapter;
 };
 
+const createAzureCommunicationCallAdapterFromClientStable = async (
+  callClient: StatefulCallClient,
+  callAgent: CallAgent,
+  locator: CallAdapterLocator
+): Promise<CallAdapter> => {
+  const deviceManager = (await callClient.getDeviceManager()) as StatefulDeviceManager;
+  return new AzureCommunicationCallAdapter(callClient, locator, callAgent, deviceManager);
+};
+
+let createAzureCommunicationCallAdapterFromClientToExport: (
+  callClient: StatefulCallClient,
+  callAgent: CallAgent,
+  locator: CallAdapterLocator,
+  /* @conditional-compile-remove(rooms) */ options?: { roleHint?: Role }
+) => Promise<CallAdapter> = createAzureCommunicationCallAdapterFromClientStable;
+
+/* @conditional-compile-remove(rooms) */
 /**
- * Create a {@link CallAdapter} using the provided {@link StatefulCallClient}.
+ * Create a {@link CallAdapter} using the provided {@link StatefulCallClient} and a provided role
+ * to hint the user of their role specific for Rooms calls.
  *
  * Useful if you want to keep a reference to {@link StatefulCallClient}.
  * Consider using {@link createAzureCommunicationCallAdapter} for a simpler API.
  *
- * @public
+ * @beta
  */
-export const createAzureCommunicationCallAdapterFromClient = async (
+createAzureCommunicationCallAdapterFromClientToExport = async (
   callClient: StatefulCallClient,
   callAgent: CallAgent,
-  locator: CallAdapterLocator
+  locator: CallAdapterLocator,
+  /* @conditional-compile-remove(rooms) */ options?: { /* @conditional-compile-remove(rooms) */ roleHint?: Role }
```

Comment:
We might need a really good jsdoc here to explain how roleHint works, otherwise it might be confusing lots of developers